src/app/auth/api.http.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { ConnectionBackend, Http, Headers, Request, RequestOptions, RequestOptionsArgs, Response } from '@angular/http';
+import { Observable } from "rxjs";
+
+import { apiToken } from "../constants";
+
+import * as _ from 'lodash';
+
+/**
+ * Wrapper for Http that appends authorization headers for requests to the API backend,
+ * and redirects unauthorized responses to login page.
+ */
+@Injectable()
+export class ApiHttp extends Http {
```

Comment:
True, although with change below it could be used for any requests that require token auth, not just our API.